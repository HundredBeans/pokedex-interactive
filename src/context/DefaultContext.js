import React, { createContext } from "react";
import { ApolloClient, ApolloProvider, InMemoryCache } from "@apollo/client";
import Localbase from "localbase";

let db = new Localbase("db");
db.config.debug = false;

const client = new ApolloClient({
  uri: "https://graphql-pokeapi.vercel.app/api/graphql",
  cache: new InMemoryCache({
    typePolicies: {
      PokemonItem: {
        fields: {
          owned: {
            read: () => {
              return 10;
            },
          },
        },
      },
    },
  }),
});

export const DefaultContext = createContext();

export const DefaultContextProvider = (props) => {
  return (
    <DefaultContext.Provider value={{ db }}>
      <ApolloProvider client={client}>{props.children}</ApolloProvider>
    </DefaultContext.Provider>
  );
};
