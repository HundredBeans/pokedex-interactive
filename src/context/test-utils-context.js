import React from "react";
import { render } from "../test-utils";
import { MockedProvider } from "@apollo/client/testing";
import { PaginationContext } from "./PaginationContext";

// Custom render to provide Apollo Provider and Pagination Context Provider for Unit Testing purpose
export const customRender = (
  ui,
  { providerProps, mockQuery, ...renderOptions }
) => {
  return render(
    <MockedProvider mocks={mockQuery} addTypename={false}>
      <PaginationContext.Provider value={providerProps}>
        {ui}
      </PaginationContext.Provider>
    </MockedProvider>,
    renderOptions
  );
};
