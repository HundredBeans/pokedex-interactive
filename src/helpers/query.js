import { gql } from "@apollo/client";

// Only query used fields
export const fetchPokemonListQuery = gql`
  query pokemons($limit: Int = 2000, $offset: Int) {
    pokemons(limit: $limit, offset: $offset) {
      count
      results {
        name
        image
        owned @client
      }
    }
  }
`;

// Only query used fields
export const fetchPokemonDetails = gql`
  query pokemon($name: String!) {
    pokemon(name: $name) {
      height
      id
      moves {
        move {
          url
          name
        }
      }
      name
      sprites {
        front_default
      }
      stats {
        base_stat
        effort
        stat {
          url
          name
        }
      }
      types {
        slot
        type {
          url
          name
        }
      }
      weight
      message
    }
  }
`;
