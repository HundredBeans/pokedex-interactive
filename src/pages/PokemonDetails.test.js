import React from "react";
import PokemonDetails from "./PokemonDetails";
import { customRender } from "../context/test-utils-context";
import { fetchPokemonDetails } from "../helpers/query";
import { fireEvent, waitFor } from "@testing-library/dom";

const id = "IVYSAUR";
jest.mock("../components/Loader", () => {
  const Loader = () => <div>Loading...</div>;
  return Loader;
});
const mockPushFn = jest.fn();
// Mock react-router-dom
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockPushFn,
  }),
  useParams: () => ({
    id,
  }),
}));

// Mock localbase
jest.mock("../helpers/db", () => ({
  db: jest.fn(),
}));
const pageLimit = 2;
const pokemonListPage = 1;
const providerProps = {
  pageLimit,
  pokemonListPage,
};
const mockQuery = [
  {
    request: {
      query: fetchPokemonDetails,
      variables: {
        name: id,
      },
    },
    result: {
      data: {
        pokemon: {
          height: 10,
          id: 2,
          moves: [
            {
              move: {
                url: "https://pokeapi.co/api/v2/move/14/",
                name: "swords-dance",
                __typename: "BaseName",
              },
              __typename: "Move",
            },
          ],
          name: "ivysaur",
          sprites: {
            front_default:
              "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
            __typename: "Sprite",
          },
          stats: [
            {
              base_stat: 60,
              effort: 0,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/1/",
                name: "hp",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
            {
              base_stat: 62,
              effort: 0,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/2/",
                name: "attack",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
            {
              base_stat: 63,
              effort: 0,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/3/",
                name: "defense",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
            {
              base_stat: 80,
              effort: 1,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/4/",
                name: "special-attack",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
            {
              base_stat: 80,
              effort: 1,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/5/",
                name: "special-defense",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
            {
              base_stat: 60,
              effort: 0,
              stat: {
                url: "https://pokeapi.co/api/v2/stat/6/",
                name: "speed",
                __typename: "BaseName",
              },
              __typename: "Stat",
            },
          ],
          types: [
            {
              slot: 1,
              type: {
                url: "https://pokeapi.co/api/v2/type/12/",
                name: "grass",
                __typename: "BaseName",
              },
              __typename: "Type",
            },
            {
              slot: 2,
              type: {
                url: "https://pokeapi.co/api/v2/type/4/",
                name: "poison",
                __typename: "BaseName",
              },
              __typename: "Type",
            },
          ],
          weight: 130,
          message: "",
          __typename: "Pokemon",
        },
      },
    },
  },
];

it("renders the component (loading)", () => {
  const renderedComponent = customRender(<PokemonDetails />, {
    providerProps,
    mockQuery,
  });
  expect(renderedComponent).toMatchSnapshot("loading");
  expect(renderedComponent.getByText("Loading..."));
});

it("renders the component (data-fetched)", async () => {
  const renderedComponent = customRender(<PokemonDetails />, {
    providerProps,
    mockQuery,
  });
  await waitFor(() => {
    expect(renderedComponent).toMatchSnapshot("data-fetched");
    expect(renderedComponent.getByText(id)).toBeInTheDocument();
    expect(renderedComponent.getByText("grass")).toBeInTheDocument();
    expect(renderedComponent.getByText("poison")).toBeInTheDocument();
  });
});

it("renders the component (error)", async () => {
  const errorMessage = "Oops, Error Occured!";
  const mockQueryError = [
    {
      request: {
        query: fetchPokemonDetails,
        variables: {
          name: id,
        },
      },
      error: new Error(errorMessage),
    },
  ];
  const renderedComponent = customRender(<PokemonDetails />, {
    providerProps,
    mockQuery: mockQueryError,
  });
  await waitFor(() => {
    renderedComponent.getByText(errorMessage);
  });
  expect(renderedComponent).toMatchSnapshot("error");
  expect(renderedComponent.getByText(errorMessage)).toBeInTheDocument();
  expect(renderedComponent.getByText(errorMessage)).toBeVisible();
});

it("can open and close the popOver", async () => {
  const renderedComponent = customRender(<PokemonDetails />, {
    providerProps,
    mockQuery,
  });
  let popOverContent;
  let popoverTriggerBtn;
  await waitFor(() => {
    // Check if the popover is not open
    popOverContent = renderedComponent.getByText("Catch ivysaur");
    expect(popOverContent).not.toBeVisible();
    popoverTriggerBtn = renderedComponent.getByRole("button");
  });
  fireEvent.click(popoverTriggerBtn);
  await waitFor(() => {
    // Check if popover visible
    expect(popOverContent).toBeVisible();
  });
  fireEvent.click(renderedComponent.getByText("Cancel"));
  await waitFor(() => {
    // Check if popover is closed
    expect(popOverContent).not.toBeVisible();
  });
});

// test("catch pokemon button", async () => {
//   const renderedComponent = customRender(<PokemonDetails />, {
//     providerProps,
//     mockQuery,
//   });
//   await waitFor(() => {
//     renderedComponent.getByRole("button");
//   });
//   fireEvent.click(renderedComponent.getByRole("button"));
//   let catchButton;
//   await waitFor(() => {
//     catchButton = renderedComponent.getByText("Catch");
//   });
//   fireEvent.click(catchButton);
// });
