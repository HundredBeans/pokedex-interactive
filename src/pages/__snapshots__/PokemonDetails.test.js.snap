// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component (data-fetched): data-fetched 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div>
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the component (data-fetched): data-fetched 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="chakra-container css-nm5t63"
      >
        <div
          class="css-dvxtzn"
        >
          <div
            class="chakra-stack css-kuxzr6"
          >
            <span
              class="css-mi7bwv"
            >
              <span
                class="css-1v3sf2x"
              >
                grass
              </span>
            </span>
            <span
              class="css-mi7bwv"
            >
              <span
                class="css-1v3sf2x"
              >
                poison
              </span>
            </span>
          </div>
          <div
            class="chakra-container css-ju1cjz"
          >
            <img
              class="chakra-image__placeholder css-1tz2n1v"
              src="placeholder.png"
            />
            <hr
              aria-orientation="vertical"
              class="chakra-divider css-1aam7vk"
              role="separator"
            />
            <div
              class="css-10sffrr"
            >
              <div
                class="css-rklm6r"
              >
                <h4
                  class="chakra-heading css-1dklj6k"
                >
                  IVYSAUR
                </h4>
              </div>
              <div
                class="chakra-stat css-1mbo1ls"
              >
                <dl>
                  <dt
                    class="chakra-stat__label css-14go5ty"
                  >
                    ID
                  </dt>
                  <dd
                    class="chakra-stat__number css-1axeus7"
                  >
                    # 
                    2
                  </dd>
                </dl>
              </div>
              <div
                class="chakra-stat css-1mbo1ls"
              >
                <dl>
                  <dt
                    class="chakra-stat__label css-14go5ty"
                  >
                    Height
                  </dt>
                  <dd
                    class="chakra-stat__number css-1axeus7"
                  >
                    10
                  </dd>
                </dl>
              </div>
              <div
                class="chakra-stat css-1mbo1ls"
              >
                <dl>
                  <dt
                    class="chakra-stat__label css-14go5ty"
                  >
                    Weight
                  </dt>
                  <dd
                    class="chakra-stat__number css-1axeus7"
                  >
                    130
                  </dd>
                </dl>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-container css-3aul1p"
        >
          <h2
            class="chakra-heading css-ci02qy"
          >
            Moves List
          </h2>
          <div
            class="no-scrollbar css-iqyxvu"
          >
            <span
              class="css-1fza2wb"
            >
              <span
                class="css-g4o07i"
              >
                swords-dance
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="css-1ue797s"
      >
        <button
          class="chakra-button css-17i1qg7"
          type="button"
        >
          <img
            class="chakra-image__placeholder css-0"
          />
        </button>
      </div>
      <div
        aria-controls="popover-content-2"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-1rpneay"
        id="popover-trigger-2"
      />
      <div
        class="css-11r8ny"
        data-popper-escaped=""
        data-popper-placement="bottom"
        data-popper-reference-hidden=""
        style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
      >
        <section
          aria-describedby="popover-body-2"
          aria-labelledby="popover-header-2"
          class="chakra-popover__content css-l1sj48"
          id="popover-content-2"
          role="dialog"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-popover__header css-1x1js2i"
            id="popover-header-2"
          >
            Catch 
            ivysaur
          </header>
          <button
            aria-label="Close"
            class="css-y3dsx3"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-popover__body css-1ews2c8"
            id="popover-body-2"
          >
            Are you sure you want to catch this Pokemon?
             
            <strong>
              (Success probability 
              50
              %)
            </strong>
          </div>
          <footer
            class="chakra-popover__footer css-1brmfnp"
          >
            <div
              class="chakra-button__group css-1ezzth2"
              role="group"
            >
              <button
                class="chakra-button css-1u73en3"
                type="button"
              >
                Cancel
              </button>
              <button
                class="chakra-button css-1k6fjms"
                type="button"
              >
                Catch
              </button>
            </div>
          </footer>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-container css-nm5t63"
    >
      <div
        class="css-dvxtzn"
      >
        <div
          class="chakra-stack css-kuxzr6"
        >
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              grass
            </span>
          </span>
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              poison
            </span>
          </span>
        </div>
        <div
          class="chakra-container css-ju1cjz"
        >
          <img
            class="chakra-image__placeholder css-1tz2n1v"
            src="placeholder.png"
          />
          <hr
            aria-orientation="vertical"
            class="chakra-divider css-1aam7vk"
            role="separator"
          />
          <div
            class="css-10sffrr"
          >
            <div
              class="css-rklm6r"
            >
              <h4
                class="chakra-heading css-1dklj6k"
              >
                IVYSAUR
              </h4>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  ID
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  # 
                  2
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Height
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  10
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Weight
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  130
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-container css-3aul1p"
      >
        <h2
          class="chakra-heading css-ci02qy"
        >
          Moves List
        </h2>
        <div
          class="no-scrollbar css-iqyxvu"
        >
          <span
            class="css-1fza2wb"
          >
            <span
              class="css-g4o07i"
            >
              swords-dance
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="css-1ue797s"
    >
      <button
        class="chakra-button css-17i1qg7"
        type="button"
      >
        <img
          class="chakra-image__placeholder css-0"
        />
      </button>
    </div>
    <div
      aria-controls="popover-content-2"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="css-1rpneay"
      id="popover-trigger-2"
    />
    <div
      class="css-11r8ny"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
    >
      <section
        aria-describedby="popover-body-2"
        aria-labelledby="popover-header-2"
        class="chakra-popover__content css-l1sj48"
        id="popover-content-2"
        role="dialog"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-popover__header css-1x1js2i"
          id="popover-header-2"
        >
          Catch 
          ivysaur
        </header>
        <button
          aria-label="Close"
          class="css-y3dsx3"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-popover__body css-1ews2c8"
          id="popover-body-2"
        >
          Are you sure you want to catch this Pokemon?
           
          <strong>
            (Success probability 
            50
            %)
          </strong>
        </div>
        <footer
          class="chakra-popover__footer css-1brmfnp"
        >
          <div
            class="chakra-button__group css-1ezzth2"
            role="group"
          >
            <button
              class="chakra-button css-1u73en3"
              type="button"
            >
              Cancel
            </button>
            <button
              class="chakra-button css-1k6fjms"
              type="button"
            >
              Catch
            </button>
          </div>
        </footer>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the component (error): error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="chakra-container css-1ism20u"
      >
        <div
          class="css-hpuyrm"
        >
          <div
            class="chakra-alert css-ldiuxo"
            role="alert"
          >
            <span
              class="chakra-alert__icon css-1yfrj3e"
            >
              <svg
                class="chakra-icon css-fi0ww0"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <div
              class="chakra-alert__title css-1luzveo"
            >
              Not Found
            </div>
            <div
              class="chakra-alert__desc css-1oa6yzk"
            >
              Oops, Error Occured!
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-container css-1ism20u"
    >
      <div
        class="css-hpuyrm"
      >
        <div
          class="chakra-alert css-ldiuxo"
          role="alert"
        >
          <span
            class="chakra-alert__icon css-1yfrj3e"
          >
            <svg
              class="chakra-icon css-fi0ww0"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            class="chakra-alert__title css-1luzveo"
          >
            Not Found
          </div>
          <div
            class="chakra-alert__desc css-1oa6yzk"
          >
            Oops, Error Occured!
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the component (loading): loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div>
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
